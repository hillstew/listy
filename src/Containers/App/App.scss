*,
*:after,
*:before {
  box-sizing: border-box;
}

body {
  min-height: 100vh;
  margin: 0;
  padding: 0;
  background: linear-gradient($main-grey, #5f6b7a);
  background-repeat: no-repeat;
}

header {
  background: linear-gradient(#939faf, $main-grey);
  position: fixed;
  border: 10px solid #FFF;
  border-radius: 0 0 30px 30px;
  border-top: none;
  height: 200px;
  width: 90%;
  right: 0;
  left: 0;
  margin: 0 auto 5px auto;
}

i {
  font-family: Font Awesome 5 Free;
}

h1 {
  font-family: $main-font;
  margin: 0;
  font-size: 50px;
  position: relative;
  color: #FFF;
  left: 85%;
  top: 160px;
  width: 130px;
  height: 40px;
  text-align: center;
  background-color: $main-grey;
  text-decoration: none;
  text-shadow: 0px 4px 4px rgba(0, 0, 0, 0.6);
}

.new-note-icon {
  text-align: center;
  color: $big-plus-icon-color;
  font-size: 60px;
  text-shadow: 0px 4px 4px rgba(0, 0, 0, 0.5);

  &:hover {
    text-shadow: 4px 4px 4px rgba(0, 0, 0, 0.5);
  }
}

#add-note-link {
  display: block;
  margin: 65px auto 0 auto;
  width: 60px;
}

#title {
  text-decoration: none;
}
