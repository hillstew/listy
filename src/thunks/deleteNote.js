import { setLoading, setError, removeNote } from '../actions';
import API from '../utils/api';

export const deleteNote = (id) => {
  return async (dispatch) => {
    try {
      dispatch(setLoading(true));
      const result = await API.fetchData(`notes/${id}`, 'DELETE');
      dispatch(setLoading(false));
      dispatch(removeNote(result));
    } catch (error) {
      dispatch(setError(error));
    }
  }
}