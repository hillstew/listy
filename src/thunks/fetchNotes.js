import { setLoading, setError, setNotes } from '../actions'

export const fetchNotes = () => {
  const url = 'http://localhost:3001/api/v1/notes';
  return async (dispatch) =>  {
    try {
      dispatch(setLoading(true))
      const response = await fetch(url)
      if(!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(setLoading(false))
      const notes = await response.json()
      dispatch(setNotes(notes))
    } catch (error) {
      dispatch(setError(error.message))
    }
  }
}