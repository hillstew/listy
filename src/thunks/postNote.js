import { setLoading, setError, addNote, togglePopup } from '../actions';
import API from '../utils/api';

export const postNote = (note) => {
  const { title, issues } = note;
  return async (dispatch) => {
    try {
      dispatch(setLoading(true));
      const result = await API.fetchData('notes', 'POST', { title, issues });
      dispatch(setLoading(false));
      dispatch(addNote(result));
    } catch (error) {
      dispatch(setError(error));
    }
  }
}