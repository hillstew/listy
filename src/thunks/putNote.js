import { setLoading, setError, updateNote } from '../actions';
import API from '../utils/api';

export const putNote = (note) => {
  const { id, title, issues } = note;
  return async (dispatch) => {
    try {
      dispatch(setLoading(true));
      await API.fetchData(`notes/${id}`, 'PUT', { title, issues });
      dispatch(updateNote(note));
    } catch (error) {
      dispatch(setError(error));
    }
    dispatch(setLoading(false));
  }
}