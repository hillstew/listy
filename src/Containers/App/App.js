import React, { Component, Fragment } from 'react'
import { Route, Switch, Redirect, withRouter } from 'react-router-dom'
import NotesSection from '../NotesSection/NotesSection'
import NoteForm from '../NoteForm/NoteForm'
import AlternateScreen from '../../Components/AlternateScreen/AlternateScreen'
import { connect } from 'react-redux'
import { fetchNotes } from '../../thunks/fetchNotes'
import { Header } from '../../Components/Header/Header'

class App extends Component {

  componentDidMount() {
    this.props.fetchNotes()
  }

  render() {
    const { notes, loading, history } = this.props
    return (
      <div>
        <Fragment>
          <Header/>
          {loading && <AlternateScreen text='Loading notes...' />}
          {!loading && 
            <div>
              <Switch>
                <Route exact path="/" component={NotesSection} />
                <Route exact path="/new-note" component={NotesSection} />
                <Route exact path="/notes/:id" component={NotesSection} />
                <Route path='*' render={() => <AlternateScreen text='404 Page Not Found' />} />
                <Route path='/not-found' render={() => <AlternateScreen text='404 Page Not Found' />} />
              </Switch>
              <Route path="/new-note" render={() => <NoteForm history={history} />} />
              <Route
                path="/notes/:id"
                render={({ match }) => {
                  const { id } = match.params
                  const note = notes.find((note) => note.id === id)
                  if (note) {
                    return <NoteForm history={history} {...note} />
                  } else {
                    return <Redirect to={'/not-found'} />
                  }
                }}
              />
            </div>
          }
        </Fragment>
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  notes: state.notes,
  loading: state.loading,
})

const mapDispatchToProps = (dispatch) => ({
 fetchNotes: () => dispatch(fetchNotes()),
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
