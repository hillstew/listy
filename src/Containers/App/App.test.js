import React, { Component } from 'react'
import App from './App'
import { shallow } from 'enzyme'
import { fetchNotes } from '../../thunks/fetchNotes'


describe('App', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<App />)
  })

  it('should match the correct snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('componentDidMount', () => {
    it('should call fetchNotes', async () => {})
  })
})
