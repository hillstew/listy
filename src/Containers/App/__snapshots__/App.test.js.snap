// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the correct snapshot 1`] = `
<div>
  <Header />
  <div>
    <Switch>
      <Route
        component={[Function]}
        path="/"
      />
      <Route
        path="*"
        render={[Function]}
      />
      <Route
        path="/not-found"
        render={[Function]}
      />
    </Switch>
    <Route
      component={[Function]}
      path="/new-note"
    />
    <Route
      path="/notes/:id"
      render={[Function]}
    />
  </div>
</div>
`;
